src/shippingState.js

```diff
@@ -0,0 +1,271 @@
+import { appState } from './shared.js';
+import { specimenCollection } from './tubeValidation.js';
+import { siteSpecificLocationToConceptId } from './shared.js';
+import conceptIds from './fieldToConceptIdMapping.js';
+
+/**
+ * Update all shipping state variables in startShipping to provide access throughout the shipping dashboard
+ * @param {object} availableCollectionsObj - the available specimen collections in the 'Available Collections' table
+ * @param {array<string>} availableLocations - the list of locations for the healthcare provider
+ * @param {array<object>} allBoxesList - the list of boxes for the healthcare provider
+ * @param {array} finalizedSpecimenList - the list of finalized specimens for the healthcare provider
+ * @param {string} userName - the name of the logged in user
+ */
+export const setAllShippingState = (availableCollectionsObj, availableLocations, allBoxesList, finalizedSpecimenList, userName) => {
+    const boxesByProviderList = filterUnshippedBoxes(allBoxesList);
+    const boxesByLocationList = filterBoxListBoxesByLocation(boxesByProviderList);
+    const providerBoxesObj = createBoxAndBagsObj(boxesByProviderList); // provider-specific data in the 'select boxes to ship' section
+    const providerBoxWithSpecimenData = addSpecimenDataToDetailBox(providerBoxesObj, finalizedSpecimenList);
+    const detailedProviderBoxes = addBoxDataToDetailBox(providerBoxWithSpecimenData, boxesByProviderList);
+    const detailedLocationBoxes = filterDetailBoxesByLocation(detailedProviderBoxes);
+
+    appState.setState({
+        allBoxesList: allBoxesList,
+        availableCollectionsObj: availableCollectionsObj,
+        availableLocations: availableLocations,
+        boxesByLocationList: boxesByLocationList,
+        detailedLocationBoxes: detailedLocationBoxes,
+        detailedProviderBoxes: detailedProviderBoxes,
+        finalizedSpecimenList: finalizedSpecimenList,
+        userName: userName,
+    });
+}
+
+export const updateShippingStateSelectedLocation = (selectedLocation) => {
+    appState.setState({
+        selectedLocation: selectedLocation,
+    });
```

Comment:
Can be shorter:
```suggestion
    appState.setState({selectedLocation});
```
Same for other locations.

Reply:
Updated all single-line instances